contracts/token/ERC721/ERC721.sol

```diff
@@ -260,13 +270,19 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata {
      * Emits a {Transfer} event.
      */
     function _mint(address to, uint256 tokenId) internal virtual {
-        require(to != address(0), "ERC721: mint to the zero address");
-        require(!_exists(tokenId), "ERC721: token already minted");
+        if (to == address(0)) {
+            revert ERC721InvalidReceiver(address(0));
+        }
+        if (_exists(tokenId)) {
+            revert ERC721InvalidSender(address(0));
```

Comment:
Is the use of this error here by design? I would've expected a separate error. It kind of makes sense, in the sense that minting is like a transfer from 0 and 0 is not the owner, but in those cases we use `IncorrectOwner`.

Reply:
This one is here even before the `IncorrectOwner` addition I think, so I didn't ask myself this question back then haha. Anyway, I somewhat agree with you but I have the problem that `IncorrectOwner` has a documented requirement saying that it `MUST be used when sender is not owner`.

Here, I think the current owner is the `address(0)` and `from` is also `address(0)`. Not to mention that `address(0)` is not a valid owner (specified in EIP-712).

I would stick to `InvalidSender` here because IMO what makes this an invalid `_mint` operation is that it can't be "sent" from the `address(0)`.

Reply:
I am still not convinced by this current error but I think we can review it when we do the ERC-721 hooks refactor.